Managers/AudioMan.h

```diff
@@ -298,38 +261,72 @@ namespace RTE {
+		/// Starts playing a certain sound file with a certain attenuation for a certain player.
 		/// </summary>
-		void StopMusic();
+		/// <param name="filePath">The path to the sound file to play.</param>
+		/// <param name="attenuation">Distance attenuation scalar: 0 = full volume, 1.0 = max distant, but not silent.</param>
+		/// <param name="player">Which player to play the SoundContainer's sounds for, -1 means all players.</param>
+		/// <returns></returns>
+		SoundContainer *PlaySound(const char *filePath, float attenuation, int player) { return PlaySound(filePath, player, attenuation, 0, PRIORITY_NORMAL, -1); }
+
+		/// <summary>
+		/// Starts playing a certain sound file with various configuration settings.
+		/// </summary>
+		/// <param name="filePath">The path to the sound file to play.</param>
+		/// <param name="attenuation">Distance attenuation scalar: 0 = full volume, 1.0 = max distant, but not silent.</param>
+		/// <param name="player">Which player to play the SoundContainer's sounds for, -1 means all players.</param>
+		/// <param name="loops">The number of times to loop the SoundContainer's sounds. 0 means play once. -1 means play infinitely until stopped.</param>
+		/// <param name="priority">The priority of this sound from 256 (lowest) to 0 (highest). Higher priority sounds are more likely to be heard.</param>
+		/// <param name="pitchOrAffectedByGlobalPitch">
+		/// The pitch to play this SoundContainer's at where 1 is unmodified frequency and each multiple of 2 is an octave up or down.
+		/// -1 means the SoundContainer will be affected by global pitch instead of setting handling its pitch manually. Defaults to -1.
+		/// </param>
+		/// <returns>Returns the new sound object being played. OWNERSHIP IS TRANSFERRED!</returns>
+		SoundContainer *PlaySound(const char *filePath, float attenuation, int player, int loops, int priority, double pitchOrAffectedByGlobalPitch);
+
+		/// <summary>
+		/// Starts playing the next sample of a certain SoundContainer for a certain player.
+		/// </summary>
+		/// <param name="pSound">Pointer to the SoundContainer to start playing. Ownership is NOT transferred!</param>
+		/// <param name="attenuation">Distance attenuation scalar: 0 = full volume, 1.0 = max distant, but not silent. Defaults to 0.</param>
+		/// <param name="player">Which player to play the SoundContainer's sounds for, -1 means all players. Defaults to -1.</param>
+		/// <param name="priority">The priority of this sound - higher priority sounds are more likely to be heard. -1 means it'll use the SoundContainer's value. Defaults to -1.</param>
+		/// <param name="pitch">/// The pitch to play this SoundContainer's at where 1 is unmodified frequency and each multiple of 2 is an octave up or down. Defaults to 1.</param>
+		/// <returns>Whether or not playback of the Sound was successful.</returns>
+		bool PlaySound(SoundContainer *pSoundContainer, float attenuation = 0.0, int player = -1, int priority = -1, double pitch = 1);
```

Comment:
All the parameter shuffling is terrible.

In `SoundContainer::Play(attenuation, player) ` you're calling `g_AudioMan.PlaySound(this, player, attenuation)` when in reality the parameter order is `g_AudioMan.PlaySound(this, attenuation, player)`.

This is just confusing and bad and likely messed up in a bunch of places.
Try to keep the order as it was originally, or at least set everything up so `player` is the first parameter (or second if a pointer is needed).

Reply:
See above comments, but I definitely forgot to fix it because cpp happily autoconvernts and won't tell me if I'm putting the wrong type in the wrong place :(

Reply:
Messy, but fair enough, but still messy.
Make sure to thoroughly check all calls to these methods so nothing is mixed up.
Also when you do a followup in the data repo check if anything lua uses these and correct them as well.

Reply:
All checked. Thankfully everything in the cpp uses SoundContainer::Play so it's all nice and contained.